### PR TITLE
Add support for DictionarySchema in code model and Add test case for report.json

### PR DESCRIPTION
**Pull request by sacheu**
* Support  DictionarySchema in code model 
* Generate code for report.json testserver swagger
* xcode project changes for new test cases

### CODE REVIEW

**Comment by sacheu** [comment]
Hand written test case is AutoRestReportTest.swift
all code under test/integration/generated/report is generated

File: test/integration/generated/report/AutoRestReport.xcodeproj/AutoRestReport_Info.plist [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<plist version="1.0">
+<dict>
+  <key>CFBundleDevelopmentRegion</key>
+  <string>en</string>
+  <key>CFBundleExecutable</key>
+  <string>$(EXECUTABLE_NAME)</string>
+  <key>CFBundleIdentifier</key>
```
Comment (tjprescott) [inline test/integration/generated/report/AutoRestReport.xcodeproj/AutoRestReport_Info.plist:8]: So AutoRestReport is a generated library with test cases?

File: test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Scheme
+   LastUpgradeVersion = "9999"
+   version = "1.3">
+   <BuildAction
```
Comment (tjprescott) [inline test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme:5]: I think this scheme is not needed. The default `swift package generate-xcodeproj` makes a bunch of schemes that we remove. I usually remove all the schemes that that command generates and just re-add the one that, in this case, would be `AutoRestReport`.

File: AutorestSwiftTest/AutoRestReportTest.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import AzureCore
+import AutoRestReport
+
+class AutoRestReportTest: XCTestCase {
```
Comment (tjprescott) [inline AutorestSwiftTest/AutoRestReportTest.swift:32]: So it looks like we are testing out ability to retrieve the report... but are we actually doing anything with the results?

**Review by tjprescott — CHANGES_REQUESTED** [review]
Tweaking the auto-generated schemes. If we can do that in an automated way, that would be great.

File: AutorestSwiftTest/AutoRestReportTest.swift (hunk 3 above)
Comment (sacheu) [reply AutorestSwiftTest/AutoRestReportTest.swift:32]: Yes, getting the report is the first step. Need to follow the rest of the steps from Laurent.

File: test/integration/generated/report/AutoRestReport.xcodeproj/AutoRestReport_Info.plist (hunk 1 above)
Comment (sacheu) [reply test/integration/generated/report/AutoRestReport.xcodeproj/AutoRestReport_Info.plist:8]: Yes, the test case is to retrieve the report.

File: AutorestSwiftTest/AutoRestReportTest.swift (hunk 3 above)
Comment (tjprescott) [reply AutorestSwiftTest/AutoRestReportTest.swift:32]: Do those steps take place in here or are they an outside thing?

Comment (sacheu) [reply AutorestSwiftTest/AutoRestReportTest.swift:32]: Outside of this test case.

File: test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme (hunk 2 above)
Comment (sacheu) [reply test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme:5]: Removed other schemes that we don't need.

File: test/integration/generated/body-file/AutoRestSwaggerBatFile.xcodeproj/xcshareddata/xcschemes/AutoRestSwaggerBatFile-Package.xcscheme [hunk 4]
```diff
@@ -1,24 +1,58 @@
...
-    selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-    shouldUseLaunchSchemeArgsEnv = "YES"
-    codeCoverageEnabled = "NO">
-    <Testables>
-    </Testables>
-  </TestAction>
+<Scheme
+   LastUpgradeVersion = "9999"
+   version = "1.3">
+   <BuildAction
```
Comment (tjprescott) [inline test/integration/generated/body-file/AutoRestSwaggerBatFile.xcodeproj/xcshareddata/xcschemes/AutoRestSwaggerBatFile-Package.xcscheme:5]: This scheme should also go away. There should only be an `AutoRestSwaggerBatFile.xcscheme`

File: test/integration/generated/head/AutoRestHeadTest.xcodeproj/xcshareddata/xcschemes/AutoRestHeadTest-Package.xcscheme [hunk 5]
```diff
@@ -1,24 +1,58 @@
...
-    selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-    shouldUseLaunchSchemeArgsEnv = "YES"
-    codeCoverageEnabled = "NO">
-    <Testables>
-    </Testables>
-  </TestAction>
+<Scheme
+   LastUpgradeVersion = "9999"
+   version = "1.3">
+   <BuildAction
```
Comment (tjprescott) [inline test/integration/generated/head/AutoRestHeadTest.xcodeproj/xcshareddata/xcschemes/AutoRestHeadTest-Package.xcscheme:5]: Same.

File: test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme (hunk 2 above)
Comment (tjprescott) [reply test/integration/generated/report/AutoRestReport.xcodeproj/xcshareddata/xcschemes/AutoRestReport-Package.xcscheme:5]: This one should be removed as well. There should only be an `AutoRestReport.xcscheme` which `generate-xcodeproj` doesn't create. That's why you remove all the ones it does create and then add this one. I can show you via screenshare if you want.

**Review by tjprescott — CHANGES_REQUESTED** [review]
The schemes still look off to me.

File: AutorestSwiftTest/AutoRestReportTest.swift (hunk 3 above)
Comment (sacheu) [reply AutorestSwiftTest/AutoRestReportTest.swift:32]: Actually, I was wrong. i look at the test report in Python side, it `print_report` after it gets the report https://github.com/Azure/autorest.python/blob/autorestv3/test/vanilla/AcceptanceTests/test_zzz.py.

**Review by tjprescott — APPROVED** [review]

**Pull request merged**